homeworks/franktoth/6_testing/test_bankaccount.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from bankaccount import BankAccount
+
+
+# Fixture to create BankAccounts with different initial balances
+@pytest.fixture(scope="class")
+def accounts():
```

Comment:
Létrehozni több objektumot egy fixture-ben technikailag lehetséges, de igazából ezzel is saját magad szivatod.
Egyrészt pl a deposit-nál tök fölöslegesen adsz meg 3 objektumot a fixture-ből, csak 1-re lenne szükséged, ezért is választod ki az első elemet a 22. soron. 
az invalid_account miatt a tesztek le sem futnak egyébként, mert ahogy átpasszolod a teszteknek a fixture-t, lefut ez a -250 balance-szal létrehozandó objektum a 10. soron, és minden elhasal ugyanazzal a hibával:

```
ERROR homeworks/franktoth/6_testing/test_bankaccount.py::TestBankAccount::test_deposit[50-550-None] - ValueError: Initial balance cannot be negative.
ERROR homeworks/franktoth/6_testing/test_bankaccount.py::TestBankAccount::test_deposit[0-330-None] - ValueError: Initial balance cannot be negative.
ERROR homeworks/franktoth/6_testing/test_bankaccount.py::TestBankAccount::test_deposit[-30-100-ValueError] - ValueError: Initial balance cannot be negative.
ERROR homeworks/franktoth/6_testing/test_bankaccount.py::TestBankAccount::test_withdraw - ValueError: Initial balance cannot be negative.
ERROR homeworks/franktoth/6_testing/test_bankaccount.py::TestBankAccount::test_transfer - ValueError: Initial balance cannot be negative.
```

hozz létre két külön fixture-t pl bankaccount1 és bankaccount2 néven, mindkettőt pozitív értékekkel. Sokkal egyszerűbb és sokkal olvashatóbb. Ha tesztelni akarod a negatív értékkel való objektum létrehozását, hogy dob e errort, azt is megteheted, de azt akkor egy pytest.raises context manager-ben tedd.

Reply:
Hmm, valamért azt gondoltam, hogy ez így működik mint egyfajta érték definiálás a teszt elején. Az feltűnt, hogy elhasal ötször a teszt, de nem volt világos, hogy már ez is a teszt része. Szétbontottam és kivettem a negatív értéket.